packages/contentstack-audit/src/modules/entries.ts

```diff
@@ -619,12 +641,155 @@ export default class Entries {
+   * Then removes those values from entry
+   * If the entry is empty then adds the first value of the options
+   * If the entry has multiple choices and min_instances then adds that number of instances
+   * @param {Record<string, unknown>}tree Contains the path where the select field can be found
+   * @param field : It contains the content-type structure of the select field
+   * @param entry : it contains the value in the entry of select field one of the options of the CT.
+   * @returns
+   */
+  fixSelectField(tree: Record<string, unknown>[], field: SelectFeildStruct, entry: any) {
+    const { enum: selectOptions, multiple, min_instance, display_type, display_name } = field;
+
+    let missingValues;
+    let isMissingValuePresent = false;
+
+    if (multiple) {
+      let obj = this.findNotPresentSelectField(entry, selectOptions);
+      let { notPresent, filteredFeild } = obj;
+      entry = filteredFeild;
+      missingValues = notPresent;
+
+      if (min_instance && Array.isArray(entry)) {
+        const missingInstances = min_instance - entry.length;
+        if (missingInstances > 0) {
+          isMissingValuePresent = true;
+          const newValues = selectOptions.choices
+            .filter((choice) => !entry.includes(choice.value))
+            .slice(0, missingInstances)
+            .map((choice) => choice.value);
+          entry.push(...newValues);
+        }
+      } else {
+        if (entry.length === 0) {
+          isMissingValuePresent = true;
+          const defaultValue = selectOptions.choices.length > 0 ? selectOptions.choices[0].value : null;
+          entry.push(defaultValue);
+        }
+      }
+    } else {
+      const isPresent = selectOptions.choices.some((choice) => choice.value === entry);
+      if (!isPresent) {
+        missingValues = entry;
+        isMissingValuePresent = true;
+        entry = selectOptions.choices.length > 0 ? selectOptions.choices[0].value : null;
+      }
+    }
+    if (display_type && isMissingValuePresent) {
+      this.missingSelectFeild[this.currentUid].push({
+        uid: this.currentUid,
+        name: this.currentTitle,
+        display_name,
+        display_type,
+        missingValues,
+        min_instance: min_instance ?? 'NA',
+        tree,
+        treeStr: tree
+          .map(({ name }) => name)
+          .filter((val) => val)
+          .join(' ➜ '),
+        fixStatus: 'Fixed',
+      });
+    }
+    return entry;
+  }
+  /**
+   *
+   * @param field It contains the value to be searched
+   * @param selectOptions It contains the options that were added in CT
+   * @returns An Array of entry containing only the values that were present in CT, An array of not present entries
+   */
+  findNotPresentSelectField(field: any, selectOptions: any) {
+    let present = [];
+    let notPresent = [];
+
+    for (const value of field) {
+      const choice = find(selectOptions.choices, { value });
```

Comment:
@cs-raj ,Instead of using find function to find elements in selectOptions.choices, we can create a Map from selectOptions.choices for constant-time lookup. This reduces the time complexity from O(n) to O(1) for each lookup operation.
You can use Map or Set.
  const choicesMap = new Map(selectOptions.choices.map(choice => [choice.value, choice]));

  for (const value of field) {
    const choice = choicesMap.get(value);
    if (choice) {
      present.push(choice.value);
    } else {
      notPresent.push(value);
    }
  }